### DISCUSSION TITLE
Substring of Webpart ID present in Search Results in highlighted summaries

### DISCUSSION THREAD

**Discussion by Namrata99sah** [microsoft-search/pnp-modern-search#1869] [Q&A]
I have deployed the latest 4.5.3.0 PNP version to resolve the Webpart ID issue that was coming up in search result(https://github.com/microsoft-search/pnp-modern-search/pull/1561), but what I can see is the GUIDs are removed from most of the pages but for few pages it's still there. And not the full GUID, only a substring is present. I found it is happening for Quick Link webpart ID which is "c70391ea-0b10-4ee9-b2b4-006d3fcad0cd", in my search results its coming as - "b2b4-006d3fcad0cd" or "4ee9-b2b4-006d3fcad0cd". So it's substring of the complete GUID. Pls help resolve this one.

![image](https://user-images.githubusercontent.com/62616681/155075296-42a3fcb1-f06a-4acc-9ba6-66f0b88b2e18.png)

**Comment by tista-chatterjee** [comment]
Facing similar issue, please help us with quick resolution.

**Comment by Suj5432** [comment]
We also see the same issue where the webpart id is displayed in the search results and users are in doubt on what and why it appears in the results.

**Comment by FranckyC** [comment]
Hi all, it seems the content of Web Parts is indexed. Can you try to disable this behavior in the site settings:

![image](https://user-images.githubusercontent.com/5659122/155333796-b8a1890e-3452-4583-8a09-9f431e4bd2b9.png)

**Comment by Namrata99sah** [comment]
Hi @FranckyC , we use the first option while re-indexing the site i.e. "Do not index Web Parts if this site contains fine-grained permissions". Do I need to change the indexing option to 3rd one and re-index the site?

**Comment by FranckyC** [comment]
Yep, try the 3rd one

**Comment by Namrata99sah** [comment]
@FranckyC Will this hide all the content present on the page to hide? As the defination speaks anything on ASPX page will be hidden.

**Comment by FranckyC** [comment]
Yes that should hide the content as well :/ Can you try in an isolated site to first see if it would be the root cause of this issue?

**Comment by FranckyC** [comment]
An other option could be to add a result type condition just for ASPX pages and not display or format the summary using Handlebars.

**Comment by Namrata99sah** [comment]
@FranckyC Is there any way to use Regex Replace function in handlebar JS..I can see only this option as of now. I tried with changing the re-index option but no luck.

**Comment by FranckyC** [comment]
Did you try the `replace` Handlebars helper? https://github.com/helpers/handlebars-helpers#replace

Otherwise you can still register your own helper using extensibility library https://microsoft-search.github.io/pnp-modern-search/extensibility/handlebars_customizations/

**Comment by wobba** [comment]
You can expect these guids to disappear over time as a change is being rolled out soon. The fix relates to how modern pages are stored/indexed and will only apply to new pages being saved once the fix has rolled out (likely within the next month). Workaround is to strip out guids.

**Reply by Namrata99sah** [reply]
@wobba Can you share what is the change? Is it only applicable to new pages or previously created pages will also be included?

**Reply by wobba** [reply]
The change will apply to old pages if they are edited and re-published. So a re-crawl will not help, it's a change in how modern pages are stored.

**Reply by damianino1** [reply]
@wobba is it also related to my issue? [https://github.com/microsoft-search/pnp-modern-search/issues/1985](url)
If yes, does latest version includes the fix for it or still in development?
